memory/jemalloc_nodump_allocator.cc

```diff
@@ -106,45 +135,66 @@ void JemallocNodumpAllocator::Deallocate(void* p) {
 }
 
 Status JemallocNodumpAllocator::InitializeArenas() {
-  // Create arena.
-  size_t arena_index_size = sizeof(arena_index_);
-  int ret =
-      mallctl("arenas.create", &arena_index_, &arena_index_size, nullptr, 0);
-  if (ret != 0) {
-    return Status::Incomplete("Failed to create jemalloc arena, error code: " +
-                              std::to_string(ret));
-  }
-  assert(arena_index_ != 0);
+  assert(!init_);
+  init_ = true;
 
-  // Read existing hooks.
-  std::string key = "arena." + std::to_string(arena_index_) + ".extent_hooks";
-  extent_hooks_t* hooks;
-  size_t hooks_size = sizeof(hooks);
-  ret = mallctl(key.c_str(), &hooks, &hooks_size, nullptr, 0);
-  if (ret != 0) {
-    return Status::Incomplete("Failed to read existing hooks, error code: " +
-                              std::to_string(ret));
-  }
+  for (size_t i = 0; i < options_.num_arenas; i++) {
+    // Create arena.
+    arena_indexes_.emplace_back();
+    size_t arena_index_size = sizeof(arena_indexes_.back());
```

Comment:
nit: I would `push_back()` on success, it is less error-prone as you don't need to `pop_back()` on failure. Then the comment below can also be simplified (no need to clean up on failure).

The other advantage is that we're more explicit about the type, since we have to respect jemalloc's ABI:
```
unsigned arena_index;
size_t arenas_index_size = sizeof(arena_index);
int ret = mallctl("arenas.create", &arena_index, ...);
if (ret != 0) { 
  ...
}
arena_indexes_.push_back(arena_index);
```

Reply:
Good point, done.